src/components/SetValidators.tsx

```diff
@@ -1,46 +1,83 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
 import { useStep } from '@/hooks/useStep';
-import { useDeploymentPageContext } from './DeploymentPageContext';
-import { Validator } from '@/types/RollupContracts';
+import { Wallet } from '@/types/RollupContracts';
 import { getRandomWallet } from '@/utils/getRandomWallet';
-import { useEffect, useState } from 'react';
+import { AddressSchema } from '@/utils/schemas';
+import { useDeploymentPageContext } from './DeploymentPageContext';
 import { StepTitle } from './StepTitle';
 import { TextInputWithInfoLink } from './TextInputWithInfoLink';
 
+const validatorsSchema = z.object({
+  numberOfValidators: z.number().min(1).max(16),
+  addresses: z.array(AddressSchema),
+});
+type ValidatorsFormValues = z.infer<typeof validatorsSchema>;
+
 export const SetValidators = () => {
-  const [{ validators: currentValidators }, dispatch] = useDeploymentPageContext();
+  const [{ validators: savedWallets }, dispatch] = useDeploymentPageContext();
   const { nextStep, validatorFormRef } = useStep();
-  const [validatorCount, setValidatorCount] = useState<number>(currentValidators?.length || 1);
-  const [validators, setValidators] = useState<Validator[]>(
-    currentValidators || Array.from({ length: validatorCount }, getRandomWallet),
+
+  const [walletCount, setWalletCount] = useState<number>(savedWallets?.length || 1);
+  const [wallets, setWallets] = useState<Wallet[]>(
+    savedWallets || Array.from({ length: walletCount }, getRandomWallet),
   );
 
+  const {
+    handleSubmit,
+    register,
+    setValue,
+    formState: { errors },
+  } = useForm({
+    resolver: zodResolver(validatorsSchema),
+    defaultValues: {
+      numberOfValidators: walletCount,
+      addresses: wallets.map((wallet) => wallet.address),
+    },
+  });
+
   useEffect(() => {
-    setValidators((prev) => {
-      if (prev.length < validatorCount) {
-        return [...prev, ...Array.from({ length: validatorCount - prev.length }, getRandomWallet)];
-      } else {
-        return prev.slice(0, validatorCount);
-      }
-    });
-  }, [validatorCount]);
+    let newWallets = wallets;
```

Comment:
Any reason we initialize to `wallets`? If I understand correctly, we overwrite it on the next few lines